src/components/QueryEditor.tsx

```diff
@@ -100,50 +136,18 @@ const QueryClause = ({
       <div className="roamjs-query-condition-relation">
         <AutocompleteInput
           value={con.relation}
-          setValue={(e) => {
-            window.clearTimeout(debounceRef.current);
-            setConditions(
-              conditions.map((c) =>
-                c.uid === con.uid ? { ...con, relation: e } : c
-              )
-            );
-            debounceRef.current = window.setTimeout(() => {
-              setInputSetting({
-                blockUid: con.uid,
-                key: "Relation",
-                value: e,
-                index: 1,
-              });
-            }, 1000);
-          }}
+          setValue={setConditionRelation}
```

Comment:
Changes to `AutocompleteInput` made it so that we need a memoized function going in to prevent infinite loops